eng/checkout-job.yml

```diff
@@ -0,0 +1,62 @@
+### Check out job pulls down the GIT repository and packs its contents
+### into an Azure artifact for reuse by the subsequent build and test execution phases.
+### We're using two separate jobs for Windows and all other platforms due to the need
+### to use different compression formats (Windows doesn't support tar.gz and the Linux
+### and OSX build clients don't have 7-zip installed) and simplify other details such
+### as line endings.
+jobs:
+- job: 'checkout_windows'
+  displayName: 'Checkout (Windows)'
+
+  variables:
+    use_open_queue: ${{ eq(variables['System.TeamProject'], 'public') }}
+
+  pool:
+    # Public Windows Build Pool
+    ${{ if variables.use_open_queue }}:
```

Comment:
It seems that you can use the variables during template expansion time in Azure Pipelines now. Last time I did something similar it was not a case. 

Just curious - have you checked whether this is working as expected on internal project? (i.e. the pools are the internal not public there).

Reply:
During my work on reusing managed test artifacts I found out that the expression evaluator in Azure DevOps sometimes behaves weirdly in the presence of conditions of the type

<pre>
${{ if <some boolean> }}:
</pre>

as opposed to

<pre>
${{ if eq(<some boolean>, true) }}:
</pre>

After fixing this, there's not much value anymore in extracting the check into the variable so I just rolled back this aspect. I still plan to test the change on an internal pipeline though.